### ISSUE TITLE
Auto-generated blank node labels are syntactically invalid

### ISSUE THREAD

**Issue by wouterbeek** [SWI-Prolog/packages-semweb#68]
When a dataset that contains blank nodes is loaded into Semweb, blank nodes assigned auto-generated labels.  The URI representation of the file path is part of these generated labels (see below).  Unfortunately, forward slashes are not allowed in Turtle-family blank node label syntax.  This means that saving to a Turtle-family format results in a syntactically malformed file.

```prolog
?- [library(semweb/rdf11)].
?- [library(semweb/turtle)].
?- rdf_load('vocab.trig', [format(trig)]).
?- rdf(S, P, O).
S = '_:file:///home/wbeek/git/Triply/cshapes/vocab.trig1',
P = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#first',
O = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#type' ;
```

Suggestion: use an MD5 hash of the URI representation of the file path i.o. the file path itself.

**Comment by JanWielemaker** [comment]
Well, the Turtle writer will rename Prolog's blank nodes into nice and short ones.    I'm not a big fan of hashes a they make debugging hard.   I see various options: make sure they never leak through standard protocols, use an encoding that can be reverted (e.g, the url-friendly base64 variant), so
we can use portray or other tools to make them readable again or use a hash.

**Comment by wouterbeek** [comment]
Indeed, the writers fix this.  I was writing atoms (bnodes and IRIs) to N-Triples directly, but this is a recipe for disaster :)  Thanks for pointing that out.

**Comment by wouterbeek** [comment]
Sorry, I never use the predicates that implement bulk saving of datasets.  I've updated then issue to make clearer that blank node renaming is currently not implemented for streamed writers.
